clang/test/compiler-course/kolodkin_g_lab_1_var_1/kolodkin_g_lab1_test.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/LabPlugin_KolodkinGrigorii_FIIT3_ClangAST%pluginext -plugin LabPlugin_KolodkinGrigorii_FIIT3_ClangAST -fsyntax-only %s 2>&1 | FileCheck %s
+
+// CHECK: Shape
+// CHECK-NEXT: |_Fields
+// CHECK-NEXT: | |_ (no fields)
+// CHECK-NEXT: |_Methods
+// CHECK-NEXT: | |_ area (double()|public|virtual|pure)
+
+// CHECK: Circle -> Shape
+// CHECK-NEXT: |_Fields
+// CHECK-NEXT: | |_ radius (double|public)
+// CHECK-NEXT: |_Methods
+// CHECK-NEXT: | |_ area (double()|public|override)
+
+// CHECK: Rectangle -> Shape
+// CHECK-NEXT: |_Fields
+// CHECK-NEXT: | |_ width (double|public)
+// CHECK-NEXT: | |_ height (double|public)
+// CHECK-NEXT: |_Methods
+// CHECK-NEXT: | |_ area (double()|public|override)
+
```

Comment:
Please, add more tests for empty/classes structs, multiple inheritance, nested classes/structs and so on